sklearn/inspection/_h_statistic.py

```diff
@@ -0,0 +1,260 @@
+
+        h_squared_pairwise : ndarray of shape (n_pairs, ) or (n_pairs, output_dim)
+            Pairwise H-squared statistic. Useful to see which feature pair has
+            strongest relative interation (relative with respect to joint effect).
+            Calculated as numerator_pairwise / denominator_pairwise.
+
+        numerator_pairwise : ndarray of shape (n_pairs, ) or (n_pairs, output_dim)
+            Numerator of pairwise H-squared statistic.
+            Useful to see which feature pair has strongest absolute interaction.
+            Take square-root to get values on the scale of the predictions.
+
+        denominator_pairwise : ndarray of shape (n_pairs, ) or (n_pairs, output_dim)
+            Denominator of pairwise H-squared statistic (not of particular interest).
+
+    References
+    ----------
+    .. [FRI] :doi:`J. H. Friedman and B. E. Popescu,
+            "Predictive Learning via Rule Ensembles",
+            The Annals of Applied Statistics, 2(3), 916-954,
+            2008. <10.1214/07-AOAS148>`
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.ensemble import HistGradientBoostingRegressor
+    >>> from sklearn.inspection import permutation_importance, h_statistic
+    >>> from sklearn.datasets import load_diabetes
+    >>>
+    >>> X, y = load_diabetes(return_X_y=True)
+    >>> est = HistGradientBoostingRegressor(max_iter=5, max_depth=2).fit(X, y)
+    >>>
+    >>> # Get Friedman's H-squared for top m=3 predictors
+    >>> m = 3
+    >>> imp = permutation_importance(est, X, y, random_state=0)
+    >>> top_m = np.argsort(imp.importances_mean)[-m:]
+    >>> h_statistic(est, X=X, features=top_m, random_state=4)
+
+    >>> # For features (8, 2), 3.4% of the joint effect variability comes from
+    >>> # their interaction. These two features also have strongest absolute
+    >>> # interaction, see "numerator_pairwise":
+    >>> # {'feature_pair': [(3, 8), (3, 2), (8, 2)],
+    >>> # 'h_squared_pairwise': array([0.00985985, 0.00927104, 0.03439926]),
+    >>> # 'numerator_pairwise': array([ 1.2955532 ,  1.2419687 , 11.13358385]),
+    >>> # 'denominator_pairwise': array([131.39690331, 133.96210997, 323.6576595 ])}
+
+    """
+    check_is_fitted(estimator)
+
+    if sample_weight is not None:
+        sample_weight = _check_sample_weight(sample_weight, X)
+
+    # Usually, the data is too large and we need subsampling
+    if X.shape[0] > n_max:
+        row_indices = sample_without_replacement(
+            n_population=X.shape[0], n_samples=n_max, random_state=random_state
+        )
+        X = _safe_indexing(X, row_indices, axis=0)
+        if sample_weight is not None:
+            sample_weight = _safe_indexing(sample_weight, row_indices, axis=0)
+    else:
+        X = X.copy()
+
+    if features is None:
+        features = feature_indices = np.arange(X.shape[1])
+    else:
+        feature_indices = np.asarray(
+            _get_column_indices(X, features), dtype=np.intp, order="C"
```

Comment:
To we need the `order="C"` and `ravel()`? Isn't it always a 1d-array?

Reply:
Not 100% sure. It is as in _partial_dependence.py, line 664.